app/interactors/tasks/set_task_type.rb

```diff
@@ -0,0 +1,13 @@
+class Tasks::SetTaskType
+    include Interactor
+
+    def call
+        if context.type == "completed"
```

Comment:
why are you assigning params in the interactor?? on why is this change required?

Reply:
The destroy_all endpoints works for two purposes: to destroy all completed tasks or to destroy all archived tasks. When the endpoint is called the type (completed or archived) params is being sent. Based on the type, the task_params should be set to either (is_completed: true) or (is_deleted: true). To handle this logic outside the controller, I included this interactor.

Reply:
The destroy_all endpoints works for two purposes: to destroy all completed tasks or to destroy all archived tasks. When the endpoint is called the type (completed or archived) params is being sent. Based on the type, the task_params should be set to either (is_completed: true) or (is_deleted: true). To handle this logic outside the controller, I included this interactor.

Reply:
why can't we pass this from frontend as params? whatever params is passed the destroy_all just handles that. If it's completed tasks then completed otherwise archived.